Modules/_csv.c

```diff
@@ -1469,15 +1496,25 @@ csv_field_size_limit(PyObject *module, PyObject *args)
                          "limit must be an integer");
             return NULL;
         }
-        _csvstate_global->field_limit = PyLong_AsLong(new_limit);
-        if (_csvstate_global->field_limit == -1 && PyErr_Occurred()) {
-            _csvstate_global->field_limit = old_limit;
+        get_csv_state(module)->field_limit = PyLong_AsLong(new_limit);
+        if (get_csv_state(module)->field_limit == -1 && PyErr_Occurred()) {
+            get_csv_state(module)->field_limit = old_limit;
             return NULL;
         }
     }
     return PyLong_FromLong(old_limit);
 }
 
+static PyType_Slot error_slots[] = {
+    {0, NULL},
+};
+
+PyType_Spec error_spec = {
+    .name = "_csv.Error",
+    .flags = Py_TPFLAGS_DEFAULT,
+    .slots = error_slots,
```

Comment:
This slots can be removed if there have no actual slot? something like `PyModuleDef`. For example: https://github.com/python/cpython/blob/master/Modules/_bisectmodule.c#L286-L292
Pter, Do you think it worth to be enhanced? MAYBE I can create a new bpo to enhance it :)

Reply:
It would need a new BPO, yes. But I don't recommend it: if people start omitting slots in the stable ABI, their extensions will fail on older Python versions :(

Reply:
> But I don't recommend it: if people start omitting slots in the stable ABI, their extensions will fail on older Python versions :(

Oh, you are right. it's a problem~

Reply:
> It would need a new BPO, yes. But I don't recommend it: if people start omitting slots in the stable ABI, their extensions will fail on older Python versions :(

But the _csv doesn't use the limited C API right now. Well, it's up to you ;-)

Reply:
_csv doesn't use it, but if `PyType_FromSpec*` started accepting `.slots = NULL`, modules that *do* use the limited API could get the problem.
I'm thinking about a solution, but that's for a PEP, not this PR :)